extensions/assets-manager/AssetsManagerEx.cpp

```diff
@@ -494,7 +530,12 @@ void AssetsManagerEx::downloadManifest()
     if (_updateState != State::PREDOWNLOAD_MANIFEST)
         return;
 
-    std::string manifestUrl = _localManifest->getManifestFileUrl();
+    std::string manifestUrl;
+    if (_remoteManifest->isVersionLoaded()) {
+        manifestUrl = _remoteManifest->getManifestFileUrl();
+    } else {
+        manifestUrl = _localManifest->getManifestFileUrl();
+    }
```

Comment:
Let me explain why we made this change.

Since we will have a initial version file bundled with the app, the first time we come to the download process, the url of the remote manifest might be different from the one in the bundled version file.

For the original logic, the AssetsManagerEx will just download from the url of the old version manifest, which is incorrect.

With this change, it will download manifest from the url in the newly downloaded version file.